### PR TITLE
refactor: Convert Media to Kotlin 

### PR DESCRIPTION
**Pull request by david-allison**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 1]
```diff
@@ -15,1042 +15,991 @@
...
 
-
-    public void forceResync() {
-        mDb.execute("delete from media");
-        mDb.execute("update meta set lastUsn=0,dirMod=0");
-        mDb.execute("vacuum");
-        mDb.execute("analyze");
-        mDb.commit();
+    open fun forceResync() {
+        db!!.execute("delete from media")
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:693]: @KotlinCleanup("scope function on db")

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 2]
```diff
@@ -15,1042 +15,991 @@
...
+            media.add(arrayOf(f, _checksum(path), mt, 1))
+        }
+        for (f in removed) {
+            media.add(arrayOf(f, null, 0, 1))
         }
         // update media db
-        mDb.executeMany("insert or replace into media values (?,?,?,?)", media);
-        mDb.execute("update meta set dirMod = ?", _mtime(dir()));
-        mDb.commit();
+        db!!.executeMany("insert or replace into media values (?,?,?,?)", media)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:563]: @KotlinCleanup("scope function on db")

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 3]
```diff
@@ -15,1042 +15,991 @@
...
-                mDb.execute("insert into meta select dirMod, usn from old.meta");
-                mDb.commit();
-            } catch (Exception e) {
+                val sql = """insert into media
+ select m.fname, csum, mod, ifnull((select 1 from log l2 where l2.fname=m.fname), 0) as dirty
+ from old.media m
+ left outer join old.log l using (fname)
+ union
+ select fname, null, 0, 1 from old.log where type=${Consts.CARD_TYPE_LRN};"""
+                db!!.execute(sql)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:111]: @KotlinCleanup("scope function on db")

**Review by lukstbit — CHANGES_REQUESTED** [review]

**Review by lukstbit — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
Another huge one! Putting it in the local rebase-merge / retest queue even though it was based of recent code

**Pull request merged**